### ISSUE TITLE
Bug: v0.3.2 NO SOUND via External Headphones

### ISSUE THREAD

**Issue by AcidSlide** [bitgapp/eqMac#326]
**Disclaimer:** 
Please go through [existing issues](https://github.com/bitgapp/eqMac/issues?q=is%3Aopen+is%3Aissue+label%3A%22type%3A+bug%22) to avoid creating duplicates. If you found a similar issue please post a comment there. If you are uncertain your issue is related to others then create the issue. Obvious ignorance of this rule might get you blocked from posting new issues. Please tick the box below to show that you have followed the guideline.
- [x] I have checked for a similar issue and sure it hasn't been reported before.

**Describe the bug**
No sound coming out when the "External Headphones" is selected. There is a high chance this is related to the #325 issue.

Wether you start eqMac with music playing or not, music app running or not, external headphones plugged in or not, once you start playing music using an external headphones no sound is coming out.

**Steps to Reproduce**
Steps to reproduce the behaviour (feel free to change the placeholder as you need):
1. Make sure no music/audio is running
2. Close all music/audio playing apps
3. Start eqMac
4. Plug your external headphones
5. Start your music (Spotify for me)
6. Play your music
7. No sound coming out of the headphones
8. Quit eqMac
9. Sound is now playing in the headphones

**Expected behaviour**
Audio should play using external headphones.

**Setup information:**
 - Audio device used for playback: External Headphones
 - Audio transmission interface:3.5mm Jack
 - macOS Version: 10.15.5
 - eqMac Version 0.3.2
 - eqMac UI Version 1.2.0
 - eqMac Driver Version 1.1.1

**Screenshots or Console.app logs**
When plugging in or trying to play music using external headphones (switching from MacBookPro Speakers to External Headphone)
```
default	05:17:11.252527+0800	eqMac	       AggregateDevice.mm:560   Built valid aggregate 127
error	05:17:11.258039+0800	eqMac	 HALC_ProxySystem::GetObjectInfo: got an error from the server, Error: 560947818 (<private>)
error	05:17:11.258064+0800	eqMac	 HALC_ShellObject::HasProperty: there is no proxy object
error	05:17:11.259465+0800	eqMac	 HALC_ProxySystem::GetObjectInfo: got an error from the server, Error: 560947818 (<private>)
error	05:17:11.259500+0800	eqMac	 HALC_ShellObject::HasProperty: there is no proxy object
error	05:17:11.259640+0800	eqMac	 HALC_ProxySystem::GetObjectInfo: got an error from the server, Error: 560947818 (<private>)
error	05:17:11.259676+0800	eqMac	 HALC_ShellObject::HasProperty: there is no proxy object
error	05:17:11.259865+0800	eqMac	 HALC_ProxySystem::GetObjectInfo: got an error from the server, Error: 560947818 (<private>)
error	05:17:11.259922+0800	eqMac	 HALC_ShellObject::HasProperty: there is no proxy object
error	2020-06-.26008527 05:17:11+0800	eqMac	 HALC_ProxySystem::GetObjectInfo: got an error from the server, Error: 560947818 (<private>)
error	05:17:11.260132+0800	eqMac	 HALC_ShellObject::HasProperty: there is no proxy object
error	05:17:11.261822+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.261826+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.261845+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.261845+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.261857+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.261869+0800	eqMac	 AudioObjectRemovePropertyListener: no object with given ID 122
error	05:17:11.261887+0800	eqMac	 AudioObjectRemovePropertyListener: no object with given ID 122
error	05:17:11.261898+0800	eqMac	 AudioObjectRemovePropertyListener: no object with given ID 122
error	05:17:11.261928+0800	eqMac	 AudioObjectRemovePropertyListener: no object with given ID 122
default	05:17:11.800096+0800	eqMac	       AggregateDevice.mm:761   HAL notification: default input device changed
default	05:17:11.846008+0800	tccd	-[TCCDAccessIdentity staticCode]: static code for: identifier com.bitgapp.eqmac, type: 0: 0x7ff38f706740 at /Applications/eqMac.app
default	05:17:11.879352+0800	eqMac	       AggregateDevice.mm:560   Built valid aggregate 138
error	05:17:11.882639+0800	eqMac	 HALC_ProxySystem::GetObjectInfo: got an error from the server, Error: 560947818 (<private>)
error	05:17:11.882684+0800	eqMac	 HALC_ShellObject::HasProperty: there is no proxy object
error	05:17:11.883135+0800	eqMac	 HALC_ProxySystem::GetObjectInfo: got an error from the server, Error: 560947818 (<private>)
error	05:17:11.883153+0800	eqMac	 HALC_ShellObject::HasProperty: there is no proxy object
error	05:17:11.884639+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.884656+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.884695+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.884706+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.884740+0800	eqMac	 ca_verify_noerr: [AudioDeviceSetProperty(mDeviceID, NULL, 0, isInput, kAudioDevicePropertyIOProcStreamUsage, theSize, theStreamUsage), 2003332927]
error	05:17:11.884756+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.884783+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.884809+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.884835+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.884849+0800	eqMac	 ca_verify_noerr: [AudioDeviceSetProperty(mDeviceID, NULL, 0, isInput, kAudioDevicePropertyIOProcStreamUsage, theSize, theStreamUsage), 2003332927]
error	05:17:11.884911+0800	eqMac	 HAL_HardwarePlugIn_DeviceDestroyIOProcID: no device
error	05:17:11.884985+0800	eqMac	 HALPlugIn::DeviceCreateIOProcID: got an error from the plug-in routine, Error: 560947818 (<private>)
error	05:17:11.891185+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891189+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891187+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891201+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891206+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891207+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891213+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891278+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891319+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891345+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891383+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891222+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891449+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891506+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891558+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891621+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891681+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891711+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891226+0800	eqMac	 HAL_HardwarePlugIn_ObjectHasProperty: no object
error	05:17:11.891812+0800	eqMac	 AudioObjectRemovePropertyListener: no object with given ID 127
error	05:17:11.891887+0800	eqMac	 AudioObjectRemovePropertyListener: no object with given ID 127
error	05:17:11.891966+0800	eqMac	 AudioObjectRemovePropertyListener: no object with given ID 127
error	05:17:11.892086+0800	eqMac	 AudioObjectRemovePropertyListener: no object with given ID 127
default	05:17:11.898817+0800	eqMac	0x10f4b01b0 - [PID=0, throttler=0x10f4eb798] ProcessThrottler::Activity::Activity: Starting background activity / 'WebPageProxy::runJavaScriptInMainFrameScriptWorld'
default	05:17:11.899048+0800	eqMac	0x10f4b01b0 - [PID=0, throttler=0x10f4eb798] ProcessThrottler::Activity::invalidate: Ending background activity / 'WebPageProxy::runJavaScriptInMainFrameScriptWorld'
default	05:17:11.900199+0800	eqMac	0x7fe5d7831020 - [pageProxyID=10, webPageID=11, PID=1062] WebPageProxy::willPerformClientRedirectForFrame: frameID = 13
default	05:17:11.900716+0800	eqMac	0x10f4b01f0 - [PID=0, throttler=0x10f4eb798] ProcessThrottler::Activity::Activity: Starting background activity / 'WebPageProxy::runJavaScriptInMainFrameScriptWorld'
default	05:17:11.903112+0800	eqMac	0x10f4b0230 - [PID=0, throttler=0x10f4eb798] ProcessThrottler::Activity::Activity: Starting background activity / 'WebPageProxy::runJavaScriptInMainFrameScriptWorld'
```

**Additional information**
Possible related to #325

**Comment by AcidSlide** [comment]
Tested in older version, no sound issues with external headphones with v0.3.0 and v0.3.1 (while I was testing installation issues see #325)

**Comment by AcidSlide** [comment]
After installing again v0.3.2 (without going through uninstall of v0.3.1), confirmed NO SOUND using External Headphones.

So I guess it works v0.3.1 and below but not on the latest release.

**Comment by Rugroden** [comment]
@AcidSlide I had a similar issue. If you open up Midi and switch the format of your headphones and eqMac from 48000Hz to 44100Hz do you get sound through eqmac again? Not sure why, but that worked for me.

**Comment by AcidSlide** [comment]
> @AcidSlide I had a similar issue. If you open up Midi and switch the format of your headphones and eqMac from 48000Hz to 44100Hz do you get sound through eqmac again? Not sure why, but that worked for me.

@Rugroden i'll try that later, but I need to reset my copy of the source code first, I think I broke it while trying to trace the sound issue ahahaha

**Comment by AcidSlide** [comment]
@Rugroden WOAH!! It worked.. switching to 44.1khz made it work

@nodeful it actually worked switching it to 44.1khz, by the way i'm using the 'development' branch, tried the 'driver-improvement' branch but gave me too much errors ahahaha

**Comment by Rugroden** [comment]
Glad that worked for you! Occasionally with this (for me anyway) the sound will start to fuzz out for like 2 seconds but then it figures itself out with no intervention from me and it's fine. I'd be interested to know if you run into that as well @AcidSlide

**Comment by nodeful** [comment]
Sorry I haven't looked into the issue thoroughly yet, as very busy with my job this week. Glad you resolved it by playing with Sampling Rate, I won't close the issue so other people can reference. Yeah the `driver-improvement` branch is work in progress at the moment.

**Comment by AcidSlide** [comment]
No problem @nodeful I just hope it also get addressed in the new `driver-improvement` that you are doing.  I would like to keep the bit rate to 48k if possible but for now manually switching to 44.1k from time to time is doable.

**Comment by kylechadha** [comment]
@AcidSlide Thank you for figuring out it works on 0.3.1! Was reading through all the issues on this repo (as well as in eqmac2 -- confusing that v3 is just called eqmac), and was able to get it working in 10 seconds just by installing 0.3.1. For me that's a better fix than switching to 44.1k

**Comment by nodeful** [comment]
Hey everyones, I think this issue has been resolved for a while now, eqMac now supports many sampling rates without a problem, sorry for the lack of updates. I will be closing this issue for now, unless someone suggests otherwise :) Thank you for your patience
